src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -575,19 +516,36 @@ private void KeepQuarantinedOr(Address remoteAddress, Action body)
 
         #region ActorBase overrides
 
-        /// <summary>
-        /// TBD
-        /// </summary>
-        /// <returns>TBD</returns>
         protected override SupervisorStrategy SupervisorStrategy()
         {
             return new OneForOneStrategy(ex =>
             {
-                var directive = Directive.Stop;
-
-                ex.Match()
-                    .With<InvalidAssociation>(ia =>
+                Directive Hopeless(HopelessAssociation e)
```

Comment:
Local function for handling quarantine behaviors - since it started it get a tad too "big" for the inside of a `switch` statement.